### ISSUE TITLE
How Can I find the accuracy of the trained SOM

### ISSUE THREAD

**Issue by johnvasu** [JustGlowing/minisom#26]
I am using Self Organizing Map to predict anomalies. I am planning to ise miniSOM library for that. I am new to ML. I am having some doubts. I would be really grateful if you could help me to find the answers. My questions are:

1. Is that possible to define the cluster type while training the SOM?

2. Is it possible to calculate the accuracy of the trained miniSOM? Is there any inbuilt method for that?

3. Is miniSOM support k-fold cross validation?

Thanks in advance

**Comment by JustGlowing** [comment]
hi @johnvasu 

1. You can use the method `labels_map` to assign your cluster, but only after the network is trained.
2. You can compute the quantization error with method `quantization_error`. However, the accuracy of your model only depends on the problems you're trying to solve. If you're predicting anomalies, you may want to use precision and recall. I'd recommend you to use the implementation in scikit-learn.
3. MiniSom doesn't implement k-fold cross validation.

Leave a star if you life MiniSom ;-)

**Comment by johnvasu** [comment]
Thank you @JustGlowing . It's a very clear explanation. I have another doubt. Can we initialize the weights in a particular range. For an instance, if I want to initialize the weights in a range of 0 and 100.Can I achieve it using miniSOM? I found only the pca_initialization

**Comment by JustGlowing** [comment]
The only two available initialization methods are `random_weights_init` and `random_weights_init`. You can modify the weights by yourself access the attribute `_weights`.

Mind that in the current release of MiniSom the weights are normalized also during the training. I'd recommend you to download the latest version from the master if you're planning to have a custom initialization.

**Comment by johnvasu** [comment]
Okay. Thank you for your help. I am closing the issue :)

**Comment by johnvasu** [comment]
Hi sir, how can I access the attribute _weights? I tried but could not find a proper way

**Comment by JustGlowing** [comment]
Here's an example:
```
som = MiniSom(2, 2, 4)
print(som._weights)
```

**Comment by MalikOvaiz** [comment]
Hi @johnvasu , 
 Is labels_map  applied on orginal training data or on test data to associate labels to clusters ? what is the purpose of using labels(labels_map  ) in SOM ?  I have no labels with training data . . After training SOM Model ( without labels ), How can I associate labels to different clusters for prediction test observation ?
